xbmc/input/joysticks/generic/InputHandling.cpp

```diff
@@ -102,13 +102,19 @@ bool CInputHandling::OnDigitalMotion(const CDriverPrimitive& source, bool bPress
   FeatureName featureName;
   if (m_buttonMap->GetFeature(source, featureName))
   {
-    FeaturePtr& feature = m_features[featureName];
-
-    if (!feature)
-      feature = FeaturePtr(CreateFeature(featureName));
+    auto it = m_features.find(featureName);
+    if (it == m_features.end())
+    {
+      FeaturePtr feature(CreateFeature(featureName));
+      if (feature)
+      {
+        m_features.insert({featureName, std::move(feature)});
```

Comment:
`std::map<>::insert` returns a `std::pair<iterator, bool>` Maybe you could skip the extra `find`?

```
it = m_features.insert({featureName, std::move(feature)}).first;
```

Reply:
rockstar. will update tomorrow

Reply:
or
```cpp
std::tie(it, std::ignore) = m_features.insert({featureName, std::move(feature)});
```
if you do not want the extra bool

Reply:
thanks, updated